File: 1272_Remove_Interval.py
class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        res = []
        rl, rr = toBeRemoved
        for il, ir in intervals:
            if ir <= rl or il >= rr:
                res.append([il, ir])
            else:
                if il < rl:
                    res.append([il, rl])
                if rr < ir:
                    res.append([rr, ir])
        return res


class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        res = []
        rl, rr = toBeRemoved
        for il, ir in intervals:
            if il < rl:
                res.append([il, min(ir, rl)])
            if rr < ir:
                res.append([max(il, rr), ir])
        return res

File: 1272_Remove_Interval.cpp
class Solution {
public:
    vector<vector<int>> removeInterval(vector<vector<int>>& intervals, vector<int>& toBeRemoved) {
        vector<vector<int>> res;
        for (auto itv : intervals) {
            if (itv[1] <= toBeRemoved[0] || itv[0] >= toBeRemoved[1]) res.push_back(itv);
            else {
                if (itv[0] < toBeRemoved[0]) res.push_back({itv[0], toBeRemoved[0]});
                if (itv[1] > toBeRemoved[1]) res.push_back({toBeRemoved[1], itv[1]});
            }
        }
        
        return res;
    }
};
